code/modules/projectiles/guns/energy/special.dm

```diff
@@ -130,14 +132,15 @@
 
 /obj/item/weapon/gun/energy/kinetic_accelerator/equipped(mob/user)
 	. = ..()
-	attempt_reload()
+	if(!can_shoot())
+		attempt_reload()
 
 /obj/item/weapon/gun/energy/kinetic_accelerator/dropped()
 	. = ..()
 	if(!holds_charge)
 		// Put it on a delay because moving item from slot to hand
 		// calls dropped().
-		spawn(overheat_time)
+		spawn(1)

```

Comment:
When you move a KA from your belt to your hand, it calls dropped(). But there's no reason to wait more than a small amount of time, because the calls happen at the same time.